frontend/dockerfile/dockerfile_test.go

```diff
@@ -6909,6 +6911,15 @@ COPY --link foo foo
 			files:          []fstest.Applier{fstest.CreateFile("foo", []byte("foo"), 0600)},
 			expectedDigest: "sha256:9f75e4bdbf3d825acb36bb603ddef4a25742afb8ccb674763ffc611ae047d8a6",
 		},
+		{
+			// https://github.com/moby/buildkit/issues/4793
+			name: "DanglingStage",
+			dockerfile: `FROM amd64/debian:bullseye-20230109 AS dangling
+FROM amd64/debian:bullseye-20230109-slim
+`,
+			expectedDigest: "sha256:eeba8ef81dec46359d099c5d674009da54e088fa8f29945d4d7fb3a7a88c450e",
```

Comment:
Could we get these digests computed with `digest.FromBytes()` . Atm. it is hard/impossible to understand what is the expected result or where this digest is coming from.

Reply:
We can't know the manifest digest until building it

Reply:
Well, you need to know the contents of the manifest if you know its digest here.

Reply:
I think it is not specific to this PR and can be revisited in a separate issue/PR?